### ISSUE TITLE
Does this package work alongside Vivox?

### ISSUE THREAD

**Issue by Vivraan** [Macoron/whisper.unity#79]
I've set up Vivox alongside a multiplayer setup in my project, which is being deployed to Quest 2 hardware. From what I understand, Vivox uses its own libraries to access the device microphone, which can then be tested in echo and group voice channels.

My question is: does this interfere in any way with how whisper.unity handles microphone usage requests (through Unity's `Microphone` API?)

**Comment by Macoron** [comment]
whisper.unity uses [custom MicrophoneRecord script](https://github.com/Macoron/whisper.unity/blob/d359e08f1fa1c9a9d78cd1a587e6cb3be837f3df/Packages/com.whisper.unity/Runtime/Utils/MicrophoneRecord.cs) to make circular recording and run VAD. If Vivox uses their own custom logic for working with a microphone - you would need to rewrite that to make them work together with whisper.

Alternatively, whisper can work with an `AudioClip`. If Vivox can provide you final `AudioClip` - you can transcribe it and doesn't interact with a microphone directly. Check how to [transcribe audio clips in samples](https://github.com/Macoron/whisper.unity/blob/d359e08f1fa1c9a9d78cd1a587e6cb3be837f3df/Assets/Samples/1%20-%20Audio%20Clip/AudioClipDemo.cs).

As for running on Quest 2, it should work, but very slow. Check #47

**Comment by Vivraan** [comment]
Yes, I only just saw https://github.com/zhouwg/kantv/issues/64, which shows somewhat fast inference on Android.

As for the Vivox integration, Vivox supports 'echoing' its audio outputs through Unity, so it's possible to use `OnAudioFilterRead` to directly pass in float arrays of the audio data to `whisper.unity`.
